plugins/inputs/xtremio/xtremio.go

```diff
@@ -0,0 +1,388 @@
+# insecure_skip_verify = false
+`
+
+// Description will appear directly above the plugin definition in the config file
+func (xio *XtremIO) Description() string {
+	return `Gathers Metrics From a Dell EMC XtremIO Storage Array's V3 API`
+}
+
+// SampleConfig will populate the sample configuration portion of the plugin's configuration
+func (xio *XtremIO) SampleConfig() string {
+	return sampleConfig
+}
+
+func (xio *XtremIO) Init() error {
+	if xio.Username == "" {
+		return fmt.Errorf("Username cannot be empty")
+	}
+	if xio.Password == "" {
+		return fmt.Errorf("Password cannot be empty")
+	}
+	if xio.URL == "" {
+		return fmt.Errorf("URL cannot be empty")
+	}
+	if len(xio.Collectors) == 0 {
+		xio.Collectors = []string{"bbus","clusters","ssds","volumes","xms"}
+	}
+
+	availableCollectors := []string{"bbus", "clusters", "ssds", "volumes", "xms"}
+	for _, collector := range xio.Collectors {
+		if !choice.Contains(collector,availableCollectors) {
+			return fmt.Errorf("Specified Collector Isn't Supported: " + collector)
+		}
+	}
+	
+	tlsCfg, err := xio.ClientConfig.TLSConfig()
+	if err != nil {
+		return err
+	}
+
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = tlsCfg
+
+	return nil
+}
+
+func (xio *XtremIO) Gather(acc telegraf.Accumulator) error {
+	var err = xio.authenticate()
+	if err != nil {
+		return err
+	}
+	if xio.Cookie == nil {
+		return fmt.Errorf("no authentication cookie set")
+	}
+
+	var wg sync.WaitGroup
+	for _, collector := range xio.Collectors {
+		wg.Add(1)
+		go func(collector string) {
+			defer wg.Done()
+
+			resp, err := xio.call(collector)
+			if err != nil {
+				acc.AddError(err)
+				return
+			}
+
+			// Due to an inconsistency in the XtremIO API, the XMS endpoint
+			// returns a json array with XMSS as the result. Which is why this
+			// if statement here exists
+			var arr []gjson.Result
+			if collector == "xms" {
+				arr = gjson.Get(resp, "xmss").Array()
+			}else{
+				arr = gjson.Get(resp, collector).Array()
+			}
+			
+			for _, item := range arr {
+				itemSplit := strings.Split(gjson.Get(item.Raw, "href").Str, "/")
+				url := ""
+				if len(itemSplit) < 1 {
+					continue
+				}
+				url = collector + "/" + itemSplit[len(itemSplit)-1]
+
+				switch collector {
+				case "bbus":
+					wg.Add(1)
+					go xio.gatherBBUs(acc, url, &wg)
+				case "clusters":
+					wg.Add(1)
+					go xio.gatherClusters(acc, url, &wg)
+				case "ssds":
+					wg.Add(1)
+					go xio.gatherSSDs(acc, url, &wg)
+				case "volumes":
+					wg.Add(1)
+					go xio.gatherVolumes(acc, url, &wg)
+				case "xms":
+					wg.Add(1)
+					go xio.gatherXMS(acc, url, &wg)
+				default:
+					acc.AddError(fmt.Errorf("Specified Collector Isn't Supported: " + collector))
+				}
+			}
+		}(collector)
+	}
+	wg.Wait()
+
+	xio.Cookie = nil
+
+	return nil
+}
+
+func (xio *XtremIO) gatherBBUs(acc telegraf.Accumulator, url string, wg *sync.WaitGroup) {
+	defer wg.Done()
+	resp, err := xio.call(url)
+	if err != nil {
+		acc.AddError(err)
```

Comment:
`acc.AddError()` will not stop the processing here but processing will continue... Is this intended? If not, please `return` here.